cannot get it running

My brain is square and I could not get this masterpiece to run.

Because my computer said some modules (I think the crypto one and some others) are deprecated or smth a couple weeks ago when I spent about 45 minutes racking my brain and googling nonsense.

So uh, can you host it somewhere like replit or something?

Here’s the logs (don’t make any sense to me as I don’t know anything about gems and stuff
```
eg@computer another-nothing % sudo bundle
Don't run Bundler as root. Bundler can ask for sudo if it is needed, and installing your bundle as root will break thisapplication for all non-root users on this machine.
Fetching gem metadata from https://rubygems.org/.......
Resolving dependencies...
Using bundler 2.3.4
Using jwt 2.3.0
Using ruby2_keywords 0.0.5
Using sequel 5.50.0
Using rack 2.2.3
Using tilt 2.0.10
Fetching sqlite3 1.4.2
Fetching kgio 2.11.4
Fetching raindrops 0.20.0
Fetching bcrypt 3.1.16
Using rack-protection 2.1.0
Fetching eventmachine 1.2.7
Using mustermann 1.1.1
Using sinatra 2.1.0
Installing bcrypt 3.1.16 with native extensions
Installing kgio 2.11.4 with native extensions
Installing sqlite3 1.4.2 with native extensions
Installing raindrops 0.20.0 with native extensions
Installing eventmachine 1.2.7 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /Library/Ruby/Gems/2.6.0/gems/bcrypt-3.1.16/ext/mri
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby -I
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0 -r ./siteconf20220114-16522-l8p3aw.rb
extconf.rb
creating Makefile

current directory: /Library/Ruby/Gems/2.6.0/gems/bcrypt-3.1.16/ext/mri
make "DESTDIR=" clean

current directory: /Library/Ruby/Gems/2.6.0/gems/bcrypt-3.1.16/ext/mri
make "DESTDIR="
make: *** No rule to make target
`/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin21/ruby/config.h',
needed by `bcrypt_ext.o'.  Stop.

make failed, exit code 2

Gem files will remain installed in /Library/Ruby/Gems/2.6.0/gems/bcrypt-3.1.16 for inspection.
Results logged to /Library/Ruby/Gems/2.6.0/extensions/universal-darwin-21/2.6.0/bcrypt-3.1.16/gem_make.out

  /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/ext/builder.rb:99:in `run'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/ext/builder.rb:51:in `block in
make'
  /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/ext/builder.rb:43:in `each'
  /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/ext/builder.rb:43:in `make'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/ext/ext_conf_builder.rb:62:in
`block in build'
  /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/tempfile.rb:295:in `open'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/ext/ext_conf_builder.rb:29:in
`build'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/ext/builder.rb:185:in `block in
build_extension'
  /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/monitor.rb:235:in `mon_synchronize'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/ext/builder.rb:181:in
`build_extension'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/ext/builder.rb:229:in `block in
build_extensions'
  /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/ext/builder.rb:226:in `each'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/ext/builder.rb:226:in
`build_extensions'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/installer.rb:830:in
`build_extensions'
  /Library/Ruby/Gems/2.6.0/gems/bundler-2.3.4/lib/bundler/rubygems_gem_installer.rb:71:in `build_extensions'
  /Library/Ruby/Gems/2.6.0/gems/bundler-2.3.4/lib/bundler/rubygems_gem_installer.rb:28:in `install'
  /Library/Ruby/Gems/2.6.0/gems/bundler-2.3.4/lib/bundler/source/rubygems.rb:204:in `install'
  /Library/Ruby/Gems/2.6.0/gems/bundler-2.3.4/lib/bundler/installer/gem_installer.rb:54:in `install'
  /Library/Ruby/Gems/2.6.0/gems/bundler-2.3.4/lib/bundler/installer/gem_installer.rb:16:in `install_from_spec'
  /Library/Ruby/Gems/2.6.0/gems/bundler-2.3.4/lib/bundler/installer/parallel_installer.rb:186:in `do_install'
  /Library/Ruby/Gems/2.6.0/gems/bundler-2.3.4/lib/bundler/installer/parallel_installer.rb:177:in `block in worker_pool'  /Library/Ruby/Gems/2.6.0/gems/bundler-2.3.4/lib/bundler/worker.rb:62:in `apply_func'
  /Library/Ruby/Gems/2.6.0/gems/bundler-2.3.4/lib/bundler/worker.rb:57:in `block in process_queue'
  /Library/Ruby/Gems/2.6.0/gems/bundler-2.3.4/lib/bundler/worker.rb:54:in `loop'
  /Library/Ruby/Gems/2.6.0/gems/bundler-2.3.4/lib/bundler/worker.rb:54:in `process_queue'
  /Library/Ruby/Gems/2.6.0/gems/bundler-2.3.4/lib/bundler/worker.rb:91:in `block (2 levels) in create_threads'

An error occurred while installing bcrypt (3.1.16), and Bundler cannot continue.

In Gemfile:
  bcrypt
```
Actually now I’m starting to think if I should have modified a config file

--- a/.dockerignore
+++ b/.dockerignore
@@ -1,0 +1,9 @@
+node_modules
+.parcel-cache
+.vagrant
+.yarn/
+build
+data
+worksonmymachine.png
+README.md
+Vagrantfile

--- a/.gitignore
+++ b/.gitignore
@@ -4,3 +4,4 @@ node_modules
 data/
 apps/*_*
 .yarn*
+.vagrant/

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,0 +1,13 @@
+# syntax=docker/dockerfile:1
+
+FROM fedora:35
+RUN dnf install ruby ruby-devel rubygem-bundler sqlite nodejs bcrypt gcc libstdc++ g++ sqlite-devel -y
+RUN corepack enable
+WORKDIR /app
+COPY . .
+RUN mkdir ./data
+VOLUME ./data
+RUN yarn
+RUN bundle
+EXPOSE 3000
+ENTRYPOINT yarn idk

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -1,0 +1,76 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# All Vagrant configuration is done below. The "2" in Vagrant.configure
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure("2") do |config|
+  # The most common configuration options are documented and commented below.
+  # For a complete reference, please see the online documentation at
+  # https://docs.vagrantup.com.
+
+  # Every Vagrant development environment requires a box. You can search for
+  # boxes at https://vagrantcloud.com/search.
+  config.vm.box = "fedora/35-cloud-base"
+
+  # Disable automatic box update checking. If you disable this, then
+  # boxes will only be checked for updates when the user runs
+  # `vagrant box outdated`. This is not recommended.
+  # config.vm.box_check_update = false
+
+  # Create a forwarded port mapping which allows access to a specific port
+  # within the machine from a port on the host machine. In the example below,
+  # accessing "localhost:8080" will access port 80 on the guest machine.
+  # NOTE: This will enable public access to the opened port
+  config.vm.network "forwarded_port", guest: 3000, host: 3000
+  config.vm.network "forwarded_port", guest: 80, host: 8080
+
+  # Create a forwarded port mapping which allows access to a specific port
+  # within the machine from a port on the host machine and only allow access
+  # via 127.0.0.1 to disable public access
+  # config.vm.network "forwarded_port", guest: 80, host: 8080, host_ip: "127.0.0.1"
+
+  # Create a private network, which allows host-only access to the machine
+  # using a specific IP.
+  # config.vm.network "private_network", ip: "192.168.33.10"
+
+  # Create a public network, which generally matched to bridged network.
+  # Bridged networks make the machine appear as another physical device on
+  # your network.
+  # config.vm.network "public_network"
+
+  # Share an additional folder to the guest VM. The first argument is
+  # the path on the host to the actual folder. The second argument is
+  # the path on the guest to mount the folder. And the optional third
+  # argument is a set of non-required options.
+  # config.vm.synced_folder "./data", "/vagrant/data"
+
+  # Provider-specific configuration so you can fine-tune various
+  # backing providers for Vagrant. These expose provider-specific options.
+  # Example for VirtualBox:
+  #
+  # config.vm.provider "virtualbox" do |vb|
+  #   # Display the VirtualBox GUI when booting the machine
+  #   vb.gui = true
+  #
+  #   # Customize the amount of memory on the VM:
+  #   vb.memory = "1024"
+  # end
+  #
+  # View the documentation for the provider you are using for more
+  # information on available options.
+
+  # Enable provisioning with a shell script. Additional provisioners such as
+  # Ansible, Chef, Docker, Puppet and Salt are also available. Please see the
+  # documentation for more information about their specific syntax and use.
+  config.vm.provision "shell", inline: <<-SHELL
+     dnf install ruby ruby-devel rubygem-bundler sqlite nodejs bcrypt gcc libstdc++ g++ sqlite-devel -y
+     cd /vagrant
+     bundle
+     corepack enable
+     corepack prepare yarn@3.1.1 --activate
+     yarn
+     yarn idk
+  SHELL
+end

--- a/main.rb
+++ b/main.rb
@@ -237,7 +237,7 @@ post '/settings' do
   #kinda dumb but ok
   s = URI(request.referrer)
   u = checklogin(request)["user"]
-  halt 500, 'no.' unless s.host == HHH && s.path == '/settings' && !request.xhr? && request.env["HTTP_SEC_FETCH_DEST"] != 'iframe'
+  halt 500, 'no.' unless request.referrer.match?(/https?:\/\/#{HHH}\/settings/) && !request.xhr? && request.env["HTTP_SEC_FETCH_DEST"] != 'iframe'
 
   if BCrypt::Password.new(USERS.first(:username => u)[:password]) == params["pass"]
     USERS.where(:username => u).update(params.reject{|k| k.start_with?('app_')||k == 'pass'||k == 'users'||k.start_with?('uninstall_')})

